src/Request/Api/BaseRequest.php

```diff
@@ -74,7 +79,16 @@ public function addArgument(IArgument $arg): IRequest
 
             return $this;
         }
-        if ($arg instanceof IStackableArgument) {
+
+        if ($arg instanceof IStackableArrayArgument) {
+            if (!isset($this->stackableArrayArguments[$arg->getKey()])) {
+                $this->stackableArrayArguments[$arg->getKey()] = new ArgArrayCollector();
+            }
+
+            $this->stackableArrayArguments[$arg->getKey()]->addArgument($arg);
+
+            return $this;
+        } elseif ($arg instanceof IStackableArgument) {
```

Comment:
Since the three conditions do almost the same thing, maybe there could be a way to refactor it? Like some kind of `ArgumentList` object where you could just do `$argumentList->addArgument($arg);`